File: BE-FindingJob/src/main/java/com/example/befindingjob/dto/JobDTO.java
package com.example.befindingjob.dto;

import com.example.befindingjob.entity.enumm.JobType;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Set;
import java.util.stream.Collectors;

@Data
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class JobDTO {
    private Integer jobId;
    private String title;
    private String description;
    private String requirements;
    private String salary;
    private String location;
    private JobType jobType;
    private String experienceRequired;
    private String postingDate;
    private String expirationDate;
    private String companyName;
    private Set<SkillDTO> requiredSkills;

    public JobDTO(ItemJobDTO itemJobDTO) {
        this.jobId = itemJobDTO.getJobId();
        this.title = itemJobDTO.getTitle();
        this.description = itemJobDTO.getDescription();
        this.requirements = itemJobDTO.getRequirements();
        this.salary = itemJobDTO.getSalary();
        this.location = itemJobDTO.getLocation();
        this.jobType = itemJobDTO.getJobType();
        this.experienceRequired = itemJobDTO.getExperienceRequired();
        this.expirationDate = (itemJobDTO.getExpirationDate() != null) ? itemJobDTO.getExpirationDate().toString() : null;
        this.companyName = (itemJobDTO.getCompany() != null) ? itemJobDTO.getCompany().getCompanyName() : null;
        this.requiredSkills = itemJobDTO.getRequiredSkills() != null
                ? itemJobDTO.getRequiredSkills().stream().collect(Collectors.toSet())
                : null;
    }
}

File: JobHub/app/src/main/java/com/example/jobhub/dto/JobDTO.kt
package com.example.jobhub.dto

import com.example.jobhub.entity.enumm.JobType

data class JobDTO(
    var jobId: Int? = null,
    var title: String? = null,
    var companyName: String? = null,
    var description: String? = null,
    var requirements: String? = null,
    var salary: String? = null,
    var location: String? = null,
    var jobType: JobType? = null,
    var experienceRequired: String? = null,
    var postingDate: String? = null,
    var expirationDate: String? = null,
    var requiredSkills: Set<SkillDTO> = emptySet(),
)
